mlp.py

```diff
@@ -0,0 +1,121 @@
+            a_prev = a
+            weight_string = "w" + str(l)
+            bias_string = "b" + str(l)
+            w = self.params[weight_string]
+            b = self.params[bias_string]
+            a, cache = linear_forward_activation(a_prev, w, b, "relu")
+            caches[str(l)] = cache
+
+        weight_string = "w" + str(number_of_layers)
+        bias_string = "b" + str(number_of_layers)
+
+        w = self.params[weight_string]
+        b = self.params[bias_string]
+
+        al, cache = linear_forward_activation(a, w, b, "sigmoid")
+        caches[str(number_of_layers)] = cache
+
+        return (al, caches)
+
+    def model_backward(self, al, y, caches):
+        grads = {}
+        num_layers = len(self.layers) - 1
+        dal = -(y / al - (1 - y) / (1 - al))
+
+        current_cache = caches[str(num_layers)]
+        da_prev, dw, db = linear_backward_activation(dal, current_cache, "sigmoid")
+
+        weight_string = "dw" + str(num_layers)
+        bias_string = "db" + str(num_layers)
+        activation_string = "dA" + str(num_layers)
+
+        grads[weight_string] = dw
+        grads[bias_string] = db
+        grads[activation_string] = da_prev
+
+        for l in reversed(range(1, num_layers)):
+            current_cache = caches[str(l)]
+            da_prev, dw, db = linear_backward_activation(da_prev, current_cache, "relu")
+
+            weight_string = "dw" + str(l)
+            bias_string = "db" + str(l)
+            activation_string = "dA" + str(l)
+
+            grads[weight_string] = dw
+            grads[bias_string] = db
+            grads[activation_string] = da_prev
+
+        return grads
+
+    def update_parameters(self, grads, learning_rate):
+        num_of_layers = len(self.layers) - 1
+
+        for l in range(1, num_of_layers + 1):
+            weight_string_grad = "dw" + str(l)
+            bias_string_grad = "db" + str(l)
+            weight_string = "w" + str(l)
+            bias_string = "b" + str(l)
+
+            self.params[weight_string] = (
+                self.params[weight_string] - learning_rate * grads[weight_string_grad]
+            )
+            self.params[bias_string] = (
+                self.params[bias_string] - learning_rate * grads[bias_string_grad]
+            )
+
+    def predict(self, X):
+        out, _ = self.model_forward(X)
+        return out.argmax(axis=0)
+
+    def train(self, X, y, iterations=20000, learning_rate=0.1):
+        X_train, X_val, y_train, y_val = train_test_split(X, y, test_size=0.2)
+
+        # One-hot encode labels
+        encoder = OneHotEncoder(sparse_output=False)
+        y_train = encoder.fit_transform(y_train.reshape(-1, 1)).T
+        y_val = encoder.fit_transform(y_val.reshape(-1, 1)).T
+
+        X_train = X_train.T
+        X_val = X_val.T
+
+        self.initialize_parameters()
+        for epoch in range(iterations):
+            al, caches = self.model_forward(X_train)
+            grads = self.model_backward(al, y_train, caches)
```

Comment:
**Suggestion:** Use a more descriptive variable name instead of 'al' for better code readability.
```suggestion
            activations, caches = self.model_forward(X_train)
            grads = self.model_backward(activations, y_train, caches)
```